face-landmarker/index.ts

```diff
@@ -12,38 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { FaceLandmarker, FilesetResolver, DrawingUtils } from '@mediapipe/tasks-vision';
+import { FaceLandmarker, DrawingUtils } from '@mediapipe/tasks-vision';
+import FaceLandmarkWorker from 'worker-loader!./Worker';
 
-const demosSection = document.getElementById("demos");
 const imageBlendShapes = document.getElementById("image-blend-shapes");
 const videoBlendShapes = document.getElementById("video-blend-shapes");
 
-let faceLandmarker;
-let runningMode: "IMAGE" | "VIDEO" = "IMAGE";
-let enableWebcamButton: HTMLButtonElement;
-let webcamRunning: Boolean = false;
-const videoWidth = 480;
+const worker = new FaceLandmarkWorker;
+worker.onmessage = (e) => {
+  if (e?.data) {
+    const { results, mode } = e.data;
+    if (mode === 'VIDEO') {
+      drawLandmarks(results, canvasElement, video.clientWidth, video.clientHeight);
+      drawBlendShapes(videoBlendShapes, results.faceBlendshapes);
+    } else {
+      const canvas = document.querySelector('#image-output-canvas') as HTMLCanvasElement;
+      const parentEl = canvas.parentElement;
+
+      drawLandmarks(results, canvas, parentEl.clientWidth, parentEl.clientHeight);
+      drawBlendShapes(imageBlendShapes, results.faceBlendshapes);
+    }
+  }
+};
 
-// Before we can use HandLandmarker class we must wait for it to finish
-// loading. Machine Learning models can be large and take a moment to
-// get everything needed to run.
-async function runDemo() {
-  // Read more `CopyWebpackPlugin`, copy wasm set from "https://cdn.skypack.dev/node_modules" to `/wasm`
-  const filesetResolver = await FilesetResolver.forVisionTasks(
-    "https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@0.10.0/wasm"
-  );
-  faceLandmarker = await FaceLandmarker.createFromOptions(filesetResolver, {
-    baseOptions: {
-      modelAssetPath: `https://storage.googleapis.com/mediapipe-models/face_landmarker/face_landmarker/float16/1/face_landmarker.task`,
-      delegate: "GPU"
-    },
-    outputFaceBlendshapes: true,
-    runningMode,
-    numFaces: 1
-  });
-  demosSection.classList.remove("invisible");
+function createImageData(object, width?: number, height?: number) {
+  const canvasForWorker = document.createElement('canvas');
```

Comment:
Can you add some inline comments that this is used to grab byte data?

This should possibly also use an ImageBitmap on Chrome since it will be much more performant.

Reply:
@schmidt-sebastian cc @khanhlvg 
Thanks for your review and suggestion!

I change to use ImageBitmap instead ImageData but got the error below, please help me check it!

<img width="656" alt="Screenshot 2023-09-15 at 9 57 57 AM" src="https://github.com/khanhlvg/mediapipe-web-demo/assets/62412677/4ca30a78-2145-4050-86a8-17558783e8cf">